src/actions/mod.rs

```diff
@@ -98,13 +98,15 @@ impl ActionHandler {
                                                   project_path: &Path)
             -> Vec<NotificationMessage<PublishRustDiagnosticsParams>>
         {
+            let cwd = ::std::env::current_dir().unwrap();
+
             build_results
             .iter()
             .map(|(path, diagnostics)| {
                 let method = "textDocument/publishDiagnostics".to_string();
 
                 let params = PublishRustDiagnosticsParams {
-                    uri: Url::from_file_path(project_path.join(path)).unwrap(),
+                    uri: Url::from_file_path(cwd.join(path)).unwrap(),
```

Comment:
The fact is we don't know if the cwd is same as project_dir, but we know that the diagnostics results's paths are relative to cwd instead of project_dir (as least for now).

From the rls process perspective, there are only two paths, one is project_dir, one is cwd. project_dir is generally predicable. While cwd is totally implementation dependent. VSCode could start rls with cwd set to itself's cwd, or set to the currently opened directory, or could be anything user/program specified.